app/graphql/mutations/create_offer_response_mutation.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Mutations
+  class CreateOfferResponseMutation < Mutations::BaseMutation
+    argument :offer_id, ID, required: true
+    argument :intended_state, Types::IntendedStateType, required: true
+
+    argument :phone_number, String, required: false
+    argument :comments, String, required: false
+    argument :rejection_reason, String, required: false
```

Comment:
The input type here could be updated to match the database requirements. I worried that that would force us to keep multiple types up-to-date (and especially because the rejection reasons themselves are longer strings), so opted to rely on the ActiveRecord validation instead.

Reply:
**question [non-blocking]:** minimum viable type checking

Can you elaborate on your worry about marking this as required? Seems like it would be better error feedback for one to pass an invalid string here than to leave it off and then get an error about the thing being required, no? I think what you're getting at is that you _could_ have created an enum with the possible valid values, but that seemed extra so you didn't. I guess my feeling is that marking this as a required string is a good middle ground. 🤷‍♂️ 

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
Ah! The field is _not_ required, I just opted to not create an additional Enum type to capture the potential inputs given they'll also be checked at creation time.

But this is not a strong opinion, I can easily create an Enum to give quicker feedback (which in retrospect might be a huge help to the UI builder).

Reply:
OMG Jon read the code my dude. What you have here totally makes sense. I saw that validation check and assumed it was required. I see now that it's NOT required and we only validate that if present, it be a certain set of values.

In terms of building the UI, totally up to you if this enum helps. Could totally come in a follow up if you wanna keep moving.